Sync/BatchingClient.swift

```diff
@@ -97,7 +115,28 @@ open class Sync15BatchClient<T: CleartextPayloadJSON> {
             >>> succeed
     }
 
-    open func addRecords(_ records: [Record<T>]) -> Success {
+    // If in batch mode, will discard the batch if any record fails
+    open func endSingleBatch() -> Success {
+        guard !records.isEmpty else {
+            return succeed()
+        }
+
+        if let token = self.batchToken {
```

Comment:
By definition, you won't have a `batchToken` if you're doing a `singleBatch` upload — you only find out that you're in a batch when you first upload some records.